### ISSUE TITLE
Building zerorpc with and without visual studio on windows

### ISSUE THREAD

**Issue by jonaslindemann** [0rpc/zerorpc-python#159]
Hi!

For the CoderDojo community I have build a raspberry pi robot, which is controlled using zerorpc. The client run on a windows platform. This was working perfectly until I installed Python and zerorpc on a different windows laptop. Identical python-versions and libraries. However, The client crashes with:

`TypeError: ord() expected string of length 1, but memoryview found`

After hours of debugging I compared the directories of the different windows machines and noticed the following difference:

On a windows machine with visual studio installed I get this in the msgpack module:
```
2016-07-29  13:29               503 exceptions.py
2016-07-29  13:29            32 073 fallback.py
2017-02-09  00:06            50 688 _packer.cp36-win32.pyd
2017-02-09  00:06            66 048 _unpacker.cp36-win32.pyd
2016-07-29  14:25                20 _version.py
2014-02-12  17:44             1 385 __init__.py
```
On a windows machine _without_ visual studio installed I get this in the msgpack module:
```
2016-07-29  13:29               503 exceptions.py
2016-07-29  13:29            32 073 fallback.py
2016-07-29  14:25                20 _version.py
2014-02-12  17:44             1 385 __init__.py
```
With different msgpack installs the zerorpc-protocol seems to fall apart. No warnings are issues when pip installing. My suggestion is that this is more clearly commented when pip installing that there can be problems when the .pyd-libraries are not built.

**Comment by bombela** [comment]
Right its because msgpack has an automatic fallback during the build.

I did try to workaround this to still but for some reason I couldn't
reproduce the problem on linux.

Feel free to suggest an addition to the README.

**Comment by jonaslindemann** [comment]
I think it automagically works on linux as it almost allways has an integrated compiler which is used during the pip install, so you will always get the optimised versions. I didn't realise the pip-install/setups on windows found and used the integrated compiler transparently.

Thanks for the help and a great library!

**Comment by bombela** [comment]
Yes but I meant that I forced msgpack to use its fallback but and it still
worked fine. What version of Python you are using?

**Comment by jonaslindemann** [comment]
Python-3.6 32 bit from python.org.

-- Jonas

On 2017-02-10, 11:36, "François-Xavier Bourlet" <notifications@github.com<mailto:notifications@github.com>> wrote:

Yes but I meant that I forced msgpack to use its fallback but and it still
worked fine. What version of Python you are using?

**Comment by Brokray** [comment]
Hello there,
I am currently facing the same issue, which seems to be well-known. Yet, there is no explicit way of fixing it. 
@jonaslindemann , would you mind sharing with us which Visual Studio packages are installed on your machine?

**Comment by jonaslindemann** [comment]
On the machine in question I had Visual Studio 2015 Community edition installed with C++ compiler. I hope this helps.

**Comment by Brokray** [comment]
Thanks! 

Sadly, after installing Visual Studio 2017 - did not find any link to older version - with C++ compiler and reinstalled msgpack-python, nothing changed. I am currently trying to install more Visual Studio packages, such as Windows SDK (which is recommended by msgpack) but the all thing is quite unclear to me...

_Side note : I just noticed that I can not uninstall anything through pip (every uninstall just pend indefinitely), might be related?_
